htdocs/takepos/invoice.php

```diff
@@ -558,7 +558,15 @@ function fail($message)
 		}
 		if ($idoflineadded <= 0) {
 			$invoice->fetch_thirdparty();
-			$idoflineadded = $invoice->addline($prod->description, $price, $qty, $tva_tx, $localtax1_tx, $localtax2_tx, $idproduct, $customer->remise_percent, '', 0, 0, 0, '', $price_base_type, $price_ttc, $prod->type, -1, 0, '', 0, (!empty($parent_line)) ? $parent_line : '', null, '', '', 0, 100, '', null, 0);
+			$array_options = array();
+
+			// complete line by hook
+			$parameters = array('prod' => $prod);
+			$reshook=$hookmanager->executeHooks('completeTakePosAddLine', $parameters, $invoice, $action);
+			if ($reshook < 0) setEventMessages($hookmanager->error, $hookmanager->errors, 'errors');
+
+			$idoflineadded = $invoice->addline($prod->description, $price, $qty, $tva_tx, $localtax1_tx, $localtax2_tx, $idproduct, $customer->remise_percent, '', 0, 0, 0, '', $price_base_type, $price_ttc, $prod->type, -1, 0, '', 0, (!empty($parent_line)) ? $parent_line : '', null, '', '', $array_options, 100, '', null, 0);
```

Comment:
A hook must return -1 (error), 0 to say continue common code or 1 to say do not execute standard code.
So i suggest to call the $invoice->addline only if $reshook == 0.